### ISSUE TITLE
Repeating label names for shortcut edit buttons under Settings in Thorium Reader

### ISSUE THREAD

**Issue by togregg** [edrlab/thorium-reader#1913]
### Current behavior:
When you try accessing the edit button for a given keyboard shortcut in Thorium Reader, a screen reader will announce it as Edit button. If you pull up a list of the buttons with your screen reader, you see a list of edit buttons repeated.
 
### Example:
Launch Thorium Reader, select Settings, navigate down to Keyboard Shortcuts, and then select Show to expose the keyboard shortcuts for Thorium Reader. Navigate to the first edit button for a shortcut and the screen reader announces Edit button. Using JAWS, bring up the Forms Field window and then arrow down to the edit buttons for shortcuts. You will see a list of the words Edit button repeated.
 
### Expected behavior:
When you try accessing the edit button for a given keyboard shortcut in Thorium Reader, a screen reader will announce [shortcut name] edit button. If you pull up a list of the buttons with your screen reader, you see a list of shortcut names with the text “edit button” on the end of each.
 
### Screenshots:
![image](https://user-images.githubusercontent.com/92333167/216446294-e457b0df-b295-4cec-b5ea-476733c76a82.png)
Figure 1 - JAWS Forms Field windows showing non-specific labels for keyboard shortcut edit buttons

![image](https://user-images.githubusercontent.com/92333167/216446561-2146a093-03c8-416b-b184-f16c278af664.png)
Figure 2 -  NVDA Element List windows showing non-specific labels for keyboard shortcut edit buttons

![image](https://user-images.githubusercontent.com/92333167/216446753-466b1796-de96-492b-a94b-b3cce3ddf01c.png)
Figure 3 - VoiceOver rotor window showing non-specific labels for keyboard shortcut edit buttons

### Machine Details:

Windows
- Windows 10 Enterprise x64 – Version 22H2 (OS Build 19045.2486)
- JAWS 2023.2212.23
- NVDA 2022.4
- Thorium Reader v2.2.0

macOS
- macOS Monterey - v12.6.2
- VoiceOver
- Thorium Reader v2.2.0
 
### Steps to recreate:
1.     Launch screen reader
2.     Launch Thorium Reader
3.     Navigate to the Toolbar menu and then select Settings toggle button
4.     Navigate to the main region
5.     Navigate to the heading Keyboard shortcuts
6.     Move down through the shortcuts to hear the screen reader announce “edit” or “edit button”.
7.     Bring up the list of form controls
- JAWS – Insert + F5
- NVDA – Insert + F7, Tab to Type and then right/left arrow to Form Fields or Buttons 
- VoiceOver – Control + Option + U, then right arrow until you get to Form Controls

8.     Arrow down to edit buttons to see a list of the words Edit button repeated.
 
### Impact:
End users trying to navigate to a specific edit button for [SMA1] [GTA2] a given shortcut command, will not be able to use quick navigation. Additionally, it is not clear which edit button goes with each keyboard shortcut when you navigate down through the shortcuts.
 
### Recommendation(s):
Each edit button should have a unique name that reflects the keyboard shortcut it is associated with in the settings within Thorium Reader.

**Comment by danielweck** [comment]
Thank you very much for taking the time to describe the issue so clearly.
I fixed the button labels, I also added an event handler for the escape key so that both screen reader and keyboard users can quickly discard the editor (instead of looking for the cancel button). I added a visual faded style to the rows / list items of keyboard shortcuts that are not currently being edited, and this is reflected in the screen reader experience by isolating the currently-edited keyboard shortcut (aria-hidden is applied to all other list items, similar to marking HTML elements as inert when a popup modal dialog is displayed, for example).
Please try and let us know if this provides a satisfactory experience. Thank you!

**Comment by togregg** [comment]
Hi Daniel,

If a new issue occurs due to the change, should it be part of this issue or a new one should be created. Also the edit button now saying the current keyboard board shortcut too, which might be confusing.

**Comment by togregg** [comment]
Hi @danielweck,

The edit buttons now include the current keyboard keys for the shortcut in its name (might be confusing). Once I activate the edit button for a keyboard shortcut, I can access the cancel and save buttons, but have lost the ability to access the buttons to modify the shortcut itself. If I arrow down past the cancel and save buttons, the screen reader says list end and if I press the down arrow again, then I go immediately to login on the settings page.

Sincerely,

Tomas Gregg, CPACC
Accessibility Analyst
UITS Assistive Technology and Accessibility Center
812-856-4112 [Main Office]
812-856-0898 [Direct Line]
***@***.******@***.***>




From: Daniel Weck ***@***.***>
Reply-To: edrlab/thorium-reader ***@***.***>
Date: Wednesday, March 1, 2023 at 2:04 PM
To: edrlab/thorium-reader ***@***.***>
Cc: Tomas Gregg ***@***.***>, Author ***@***.***>
Subject: [External] Re: [edrlab/thorium-reader] Repeating label names for shortcut edit buttons under Settings in Thorium Reader (Issue #1913)

This message was sent from a non-IU address. Please exercise caution when clicking links or opening attachments from external sources.


Thank you very much for taking the time to describe the issue so clearly.
I fixed the button labels, I also added an event handler for the escape key so that both screen reader and keyboard users can quickly discard the editor (instead of looking for the cancel button). I added a visual faded style to the rows / list items of keyboard shortcuts that are not currently being edited, and this is reflected in the screen reader experience by isolating the currently-edited keyboard shortcut (aria-hidden is applied to all other list items, similar to marking HTML elements as inert when a popup modal dialog is displayed, for example).
Please try and let us know if this provides a satisfactory experience. Thank you!

**Comment by danielweck** [comment]
This is changing (hopefully for the better) in Thorium 3.0 which we are working on very actively. I am closing this issue as we need to work from feedback based on the refactored GUI (currently testable by downloading installers from GitHub)

**Comment by togregg** [comment]
Since the issue isn't fixed yet, I would prefer that it is left open until it is addressed with 3.0.

**Comment by togregg** [comment]
@danielweck - This issue persists in Version 3, please reopen this issue.

**Comment by danielweck** [comment]
@togregg just to confirm that this was fixed some time ago in the develop branch, not released yet (will be version 3.1.0)
